common/model/tests/test_transaction.c

```diff
@@ -49,12 +49,125 @@ void test_reset(void) {
+
+  uint64_t test_snapshot_index = 2345;
+  bool test_solid = true;
+  uint64_t test_arrival_timestamp = 1477037812737;
+
+  iota_transaction_t transaction;
+
+  // write transaction values
+  flex_trits_from_trytes(test_address, FLEX_TRIT_SIZE_243, TEST_ADDRESS_0,
+                         NUM_TRYTES_ADDRESS, sizeof(TEST_ADDRESS_0));
+  transaction_set_address(&transaction, test_address);
+  transaction_set_value(&transaction, test_value);
+  flex_trits_from_trytes(test_obsolete_tag, FLEX_TRIT_SIZE_81,
+                         TEST_OBSOLETE_TAG, NUM_TRYTES_OBSOLETE_TAG,
+                         sizeof(TEST_OBSOLETE_TAG));
+  transaction_set_obsolete_tag(&transaction, test_obsolete_tag);
+  transaction_set_timestamp(&transaction, test_timestamp);
+  transaction_set_current_index(&transaction, test_current_index);
+  transaction_set_last_index(&transaction, test_last_index);
+  flex_trits_from_trytes(test_bundle, FLEX_TRIT_SIZE_243, TEST_BUNDLE,
+                         NUM_TRYTES_BUNDLE, sizeof(TEST_BUNDLE));
+  transaction_set_bundle(&transaction, test_bundle);
+
+  flex_trits_from_trytes(test_trunk, FLEX_TRIT_SIZE_243, TEST_TRUNK,
+                         NUM_TRYTES_TRUNK, sizeof(TEST_TRUNK));
+  transaction_set_trunk(&transaction, test_trunk);
+  flex_trits_from_trytes(test_branch, FLEX_TRIT_SIZE_243, TEST_BRANCH,
+                         NUM_TRYTES_BRANCH, sizeof(TEST_BRANCH));
+  transaction_set_branch(&transaction, test_branch);
+  transaction_set_attachment_timestamp(&transaction, test_attachment_timestamp);
+  transaction_set_attachment_timestamp_lower(&transaction,
+                                             test_attachment_timestamp_lower);
+  transaction_set_attachment_timestamp_upper(&transaction,
+                                             test_attachment_timestamp_upper);
+  flex_trits_from_trytes(test_nonce, FLEX_TRIT_SIZE_81, TEST_NONCE,
+                         NUM_TRYTES_NONCE, sizeof(TEST_NONCE));
+  transaction_set_nonce(&transaction, test_nonce);
+  flex_trits_from_trytes(test_tag, FLEX_TRIT_SIZE_81, TEST_TAG_NULL,
+                         NUM_TRYTES_TAG, sizeof(TEST_TAG_NULL));
+  transaction_set_tag(&transaction, test_tag);
+
+  flex_trits_from_trytes(test_hash, FLEX_TRIT_SIZE_243, TEST_HASH,
+                         NUM_TRYTES_HASH, sizeof(TEST_HASH));
+  transaction_set_hash(&transaction, test_hash);
+  flex_trits_from_trytes(test_signature_or_message, FLEX_TRIT_SIZE_6561,
+                         TEST_SIG_1, NUM_TRYTES_SIGNATURE, sizeof(TEST_SIG_1));
+  transaction_set_signature(&transaction, test_signature_or_message);
+
+  transaction_set_snapshot_index(&transaction, test_snapshot_index);
+  transaction_set_solid(&transaction, test_solid);
+  transaction_set_arrival_timestamp(&transaction, test_arrival_timestamp);
+
+  // read transaction values
+  TEST_ASSERT_EQUAL_MEMORY(transaction_address(&transaction), test_address,
+                           sizeof(test_address));
```

Comment:
It would be awesome if you could test that the masks associated with each field are set